### DISCUSSION TITLE
How to validate a single boolean value?

### DISCUSSION THREAD

**Discussion by gabrc52** [pydantic/pydantic#9889] [Question]
I am using Pydantic extensively in my repository. 

Pydantic is good at validating models. However, there are some cases where I might want to validate a single value, in this case a boolean. There are two examples:

* I want to use a ["before" validator](https://docs.pydantic.dev/latest/concepts/validators/), which runs before Pydantic's internal parsing and validation (such as converting from string to bool), but I want to use a boolean value in my model which Pydantic has not validated yet. Other values in my model depend on this boolean. In this case, I am forced to validate the boolean manually.
* I want to parse command line arguments. Using `type=bool` with `ArgumentParser` would be a bad idea because the only falsy value it knows is an empty string (as mentioned in https://docs.python.org/3/library/argparse.html#type). It would be nice to use Pydantic for this, for consistent behavior and because why reinvent the wheel when I already have Pydantic in my repository?

The two solutions I could come up with are:

## Option 1. Make a dummy model that only has a single boolean

```py
class _BoolWrapper(BaseModel):
    value: bool

def parse_bool(value: any) -> bool:
    return _BoolWrapper.model_validate(dict(value=value)).value
```

This works but it feels hacky to make a wrapper model with only one value when all I want is to validate a single boolean.

## Option 2. Use a function from Pydantic v1

```py
from pydantic.v1.validators import bool_validator

print(bool_validator('on'))  # True
print(bool_validator('off'))  # False
print(bool_validator(0))  # False
print(bool_validator(1))  # True
print(bool_validator('true'))  # True
print(bool_validator('false'))  # False
```

This works. But I don't know if using this function from an old version risks becoming deprecated or removed... And from what I could gather this is meant to be used as part of a decorator and not standalone?

## Question

Is there a better way that I could not find or think of?

**Comment by Viicos** [comment]
You probably need [`TypeAdapter`](https://docs.pydantic.dev/latest/concepts/type_adapter/):

```python
from pydantic import TypeAdapter

ta = TypeAdapter(bool)
ta.validate_python('on')
#> True
```

**Reply by sezanzeb** [reply]
~What if I want `ta.validate_python('on')` to fail, and only `ta.validate_python(True)` to work?~

Ah, nevermind, `validate_python` has a `strict` argument:

```py
from pydantic import TypeAdapter

ta = TypeAdapter(bool)
ta.validate_python('on', strict=True)  # raises ValidationError
```
